src/ray/raylet/node_manager.cc

```diff
@@ -1982,8 +1989,16 @@ void NodeManager::ExtendActorFrontier(const ObjectID &dummy_object,
 void NodeManager::FinishAssignedActorCreationTask(const ActorID &parent_actor_id,
                                                   const TaskSpecification &task_spec,
                                                   bool resumed_from_checkpoint) {
-  // Notify the other node managers that the actor has been created.
   const ActorID actor_id = task_spec.ActorCreationId();
+  // Halt creation if parent actor is dead.
+  auto parent_actor_entry = actor_registry_.find(parent_actor_id);
+  if (parent_actor_entry != actor_registry_.end() &&
+      parent_actor_entry->second.GetState() == ActorTableData::DEAD) {
+    RAY_LOG(DEBUG) << "Halting creation of actor " << actor_id << " as parent actor "
+                   << parent_actor_id << " is dead.";
+    return;
```

Comment:
We should not exit here since now it will appear as if the child actor was created, but really it was not. And if we try to submit any tasks to it, it will hang because there won't be an actor table entry.

Instead, let's allow this function to complete and move this logic to the actor table handler for creation notifications ([here](https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L657)). We can then call `ProcessDisconnectClientMessage` there same as in the other case, if the actor is local.